core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go

```diff
@@ -133,9 +133,12 @@ func (enc EVMAutomationEncoder21) Extract(raw []byte) ([]ocr2keepers.ReportedUpk
 			common.BytesToHash(triggerW.BlockHash[:]).Hex(),
 			logExt,
 		)
-		triggerID, _ := UpkeepTriggerID(upkeepId, report.Triggers[i])
+		id, _ := UpkeepTriggerID(upkeepId, report.Triggers[i])
+		// TODO workID should exclude the checkblock number
+		workId, _ := UpkeepTriggerID(upkeepId, report.Triggers[i])
```

Comment:
@EasterTheBunny Question: how to get rid of checkblock number here? Thanks.

Reply:
You will probably need to create a new helper function for the WorkID.